test_runner/batch_others/test_wal_acceptor_async.py

```diff
@@ -0,0 +1,166 @@
+import asyncio
+import aiopg
+import random
+
+from fixtures.zenith_fixtures import WalAcceptor, WalAcceptorFactory, ZenithPageserver, PostgresFactory, Postgres
+from typing import List
+
+pytest_plugins = ("fixtures.zenith_fixtures")
+
+
+class BankClient(object):
+    def __init__(self, conn: aiopg.Connection, n_accounts=None, init_amount=None):
+        self.conn: aiopg.Connection = conn
+        self.n_accounts = n_accounts
+        self.init_amount = init_amount
+
+    async def initdb(self):
+        async with self.conn.cursor() as cur:
+            await cur.execute('DROP TABLE IF EXISTS bank_accs')
+            await cur.execute('CREATE TABLE bank_accs(uid int primary key, amount int)')
+            await cur.execute(
+                '''
+            INSERT INTO bank_accs
+            SELECT *, %s FROM generate_series(0, %s)
+            ''', (
+                    self.init_amount,
+                    self.n_accounts - 1,
+                ))
+
+            await cur.execute('CREATE TABLE bank_log(from_uid int, to_uid int, amount int)')
+
+    async def check_invariant(self):
+        async with self.conn.cursor() as cur:
+            await cur.execute('SELECT sum(amount) FROM bank_accs')
+            assert await cur.fetchone() == (self.n_accounts * self.init_amount, )
+
+    async def transfer(self, from_uid, to_uid, amount):
+        # avoid deadlocks by sorting uids
+        if from_uid > to_uid:
+            from_uid, to_uid, amount = to_uid, from_uid, -amount
+
+        async with self.conn.cursor() as cur:
+            await cur.execute('BEGIN')
+            await cur.execute('UPDATE bank_accs SET amount = amount + (%s) WHERE uid = %s', (
+                amount,
+                to_uid,
+            ))
+            assert cur.rowcount == 1
+            await cur.execute('UPDATE bank_accs SET amount = amount - (%s) WHERE uid = %s', (
+                amount,
+                from_uid,
+            ))
+            assert cur.rowcount == 1
+            await cur.execute('INSERT INTO bank_log VALUES (%s, %s, %s)', (
+                from_uid,
+                to_uid,
+                amount,
+            ))
+            await cur.execute('COMMIT')
+
+
+class WorkerStats(object):
+    def __init__(self, n_workers):
+        self.n_workers = n_workers
```

Comment:
Do we need `n_workers` after we've initialized `self.counters`? We could restore this value using `len`.

Reply:
`n_workers` is not really needed